src/nodes/constraint_aware_append/constraint_aware_append.c

```diff
@@ -78,6 +79,8 @@ can_exclude_chunk(PlannerInfo *root, EState *estate, Index rt_index, List *restr
 {
 	RangeTblEntry *rte = rt_fetch(rt_index, estate->es_range_table);
 
+	/* Need AccessShareLock to call excluded_by_constraint() */
+	LockRelationOid(rte->relid, AccessShareLock);
```

Comment:
The fact that it's not locked in this place looks like a bug. All RangeTableEntries are supposed to be locked in AccessShareLock mode very early, in upstream inheritance this is done in `expand_inherited_rtentry() -> find_inheritance_children()`. Did we somehow miss the lock in our chunk expansion code?

Reply:
Yes, it looks like it. You might be right that this could be a bigger issue with our expansion code so I will dig a bit deeper to see what I can find.